taskfile/cache_internal_test.go

```diff
@@ -0,0 +1,99 @@
+package taskfile
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/go-task/task/v3/internal/logger"
+)
+
+func ExampleWithTTL() {
+	c, _ := NewCache(os.TempDir(), WithTTL(2*time.Minute+30*time.Second))
+
+	fmt.Println(c.ttl)
+	// Output: 2m30s
+}
+
+var discardLogger = &logger.Logger{
+	Stdout: io.Discard,
+	Stderr: io.Discard,
+}
+
+func primeNewCache(t *testing.T, tempDir string, cacheOpts ...CacheOption) (*Cache, *FileNode) {
+	t.Helper()
+
+	cache, err := NewCache(tempDir, cacheOpts...)
```

Comment:
Do you use this testing helper with another folder than the.TmpDir() ?

If not then you could remove the tempdir argument and use 


```suggestion
	cache, err := NewCache(t.TempDir(), cacheOpts...)
```

Reply:
I was taking `tempDir` as an argument there because I re-use it on the next line, and calling it twice would give two different directories. I suppose I'll take it out of L32?

Reply:
Move this 

```go
        require.NoErrorf(t, err, "creating new cache in temporary directory '%s'", tempDir)
```

To

NewCache by using a fmt.Errorf to add the temporary Dir/file you were trying to create

Reply:
Something like this in NewCache

```go
         if err := os.MkdirAll(dir, 0o755); err != nil { 
                  return nil, fmt.Errorf("creating cache dir %s: %w", dir, err)
          }
```

Reply:
ea229890a392eb4615386110adecaf22478c54bf will resolve.